File: javascript/packages/flyql/snippets/recipes/eq-status.mjs
// Filter records where status field is exactly 200.
//
// FlyQL query: status = 200

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse('status = 200')

const columns = {
    status: newColumn({ name: 'status', type: 'UInt32' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)

File: python/snippets/recipes/eq_status.py
"""Filter records where status field is exactly 200.

FlyQL query: ``status = 200``
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse("status = 200")

columns = {
    "status": Column("status", "UInt32"),
}

sql = to_sql_where(result.root, columns)
print(sql)
